### ISSUE TITLE
updateCurrentUserInfoWithProfileImage stopped working between 3.0.48 and 3.0.49

### ISSUE THREAD

**Issue by simonwatt** [smilefam/SendBird-SDK-JavaScript#65]
@Alchemist85K 

I was on version 3.0.37 and upgraded to 3.0.51, I then noticed that updateCurrentUserInfoWithProfileImage is no longer working. 

I went back versions until I got to 3.0.48 at which point it started to work again, so it seems like this issue was introduced in 3.0.49.

The error callback in 3.0.49 and above (lower versions with the exact same data passed works fine) isn't helpful and only contains the following:

"{"name":"SendBirdException","code":0,"message":null}"

Note - I'm using Chrome, and the file data I'm passing works fine in 3.0.48 and earlier.

**Comment by simonwatt** [comment]
Seeing as you guys (SendBird) never actually respond to issues on here (not just this one, but most issues posted), maybe you could considering making the JavaScript SDK open source and let users make PR's to fix issues themselves?
@sf-jay-yun @sf-craig-park

**Comment by Alchemist85K** [comment]
@simonwatt 
Thank you for your interest in SendBird SDK.
So sorry the answer is late. 
Do you still have the same issue in the latest version SDK?
If you still have the same problem, tell me about problem more detail.

**Comment by simonwatt** [comment]
@Alchemist85K Thanks for the reply. Yeah, I just updated to the latest (3.0.53) and I get the same issue. As above I need to rollback as far as 3.0.48 to not get an issue.

Interestingly if I view the network requests the payload getting sent to 
https://api.sendbird.com/v3/users/[myUserId] is different.

In both cases my call to: updateCurrentUserInfoWithProfile image is the same and looks something like:
```
sb.updateCurrentUserInfoWithProfileImage(nickname, imageBlob, function(response, error) {   
//do stuff
});
```

**3.0.48 network request looks like:**
```
Request URL:https://api.sendbird.com/v3/users/[myUserId]
Request Method:PUT
Status Code:200 OK
Request Payload:
{"nickname":"my nickname","profile_url":"https://sendbird-upload.s3.amazonaws.com/84F2BC60-6F7A-41BA-8C48-96A5E804D21B/profile/62b3a3744e18676a33d36105a7e692fb1ec98a4f.jpg"}
```

**3.0.53 network request looks like:**
```
Request URL:https://api.sendbird.com/v3/users/[myUserId]
Request Method:PUT
Status Code:500 INTERNAL SERVER ERROR
Request Payload:------WebKitFormBoundarybB9jc0mooh585wYg
Content-Disposition: form-data; name="nickname"

my nickname
------WebKitFormBoundarybB9jc0mooh585wYg
Content-Disposition: form-data; name="profile_file"; filename="blob"
Content-Type: image/png

------WebKitFormBoundarybB9jc0mooh585wYg--
```


Additionally.... 3.0.48 (the one that works) is making a call to `https://api.sendbird.com/v3/storage/profile_image` before the call to the `https://api.sendbird.com/v3/users/[myUserId]` end point. The call to `https://api.sendbird.com/v3/storage/profile_image` is getting passed what looks to be the same data that is getting passed to `https://api.sendbird.com/v3/users/[myUserId]` in 3.0.53.

**Comment by Alchemist85K** [comment]
@simonwatt 
The file upload method used in `updateCurrentUserInfoWithProfileImage` has changed. so It different from the previous version.
However, I did not find problem that you provided.
Could you please send e-mail the used image file and code. james@sendbird.com

**Comment by simonwatt** [comment]
@Alchemist85K 
The project it's currently used on is pretty complex, so I'll try and make a more simplified example before sending you any code.

Before I do that though, is there any chance the change to updateCurrentUserInfoWithProfileImage was to do with handling exif orientation data on the image? I ask because I'm doing a few things after a user selects the file to rotate the image, as otherwise images taken on some phones are displaying at the wrong orientation. So if SendBird is now handling that I can remove that code and there is a chance the issue goes away.

**Comment by simonwatt** [comment]
@Alchemist85K I just spent some time trying to narrow this down further; it turns out that if I pass a `File `object to `updateCurrentUserInfoWithProfileImage `then it works fine for me, even in 3.0.53. Whereas if I pass a `Blob ` object then it's working in <=3.0.48 but not higher versions.

The reason I'm passing a Blob (and not the originally selected File) is that this is what I'm creating after a I manipulate the image based on it's exif orientation.

Any chance we can get support back for Blob in the next version or do I have to find a way to pass a File object instead? Alternatively if you guys handle the exif orientation data on Profile Image upload I wouldn't have to do this in the first place...

**Comment by Alchemist85K** [comment]
@simonwatt 
We've no plans to add that function yet.
We recommend that you change the orientation of the image when you receive the message and use `data` field of the message to store additional data of image.
I will share your good feedback with the team and discuss them.
Thank you.

**Comment by Alchemist85K** [comment]
@simonwatt 
You can use blob in `updateCurrentUserInfoWithProfileImage()`. 
Please, could you send an email to me with your issue image file and detail code?
james@sendbird.com

**Comment by simonwatt** [comment]
@Alchemist85K 
The project I'm using it on is quite complex so instead I have made a fork of your sample project'(the web-sample demo) project to demonstrate the issue: https://github.com/simonwatt/SendBird-JavaScript

Currently I have  web-sample/chat.html referencing 3.0.55: `<script src="static/lib/SendBird3055.min.js"></script>`, the 'Save as Blob' button that I have added fails, change that back to the original version the sample uses (3.0.35)  `<script src="static/lib/SendBird.min.js"></script>` and the Blob is now saving correctly when I press that button.

To do the test: login, then scroll to the very bottom left (below 'username') and you will see I have added 2 buttons: 'Save as File' and 'Save as Blob'. Change the SDK version as appropriate (file is referenced in chat.html) to replicate the issue. By default it's pointing to 3.0.55 so it is breaking.
